addons/website_portal_sale/models/sale_order.py

```diff
@@ -9,11 +9,16 @@ class SaleOrder(models.Model):
     _inherit = 'sale.order'
 
     website_url = fields.Char('Website URL', compute='_website_url', help='The full URL to access the document through the website.')
+    payment_transaction_count = fields.Integer(compute='_compute_payment_transaction_count', string="Number of payment transactions")
 
     def _website_url(self):
         for so in self:
             so.website_url = '/my/orders/%s' % (so.id)
 
+    def _compute_payment_transaction_count(self):
+        for order in self:
```

Comment:
Use a read group, even if normally we use it only on a form view, do not forget that strange people code in odoo.